### PR TITLE
Bugfix + segwit utilities

### PR DESCRIPTION
**Pull request by the9ull**
Hello,
I worked on another implementation of sewit in python-bitcoinlib. The most of the code is overlapped to your implementation.
I can contribute with some utility functions useful tu build p2wsh transactions.
Regards

### CODE REVIEW

File: bitcoin/core/script.py [hunk 1]
```diff
@@ -700,7 +700,7 @@ def is_witness_v0_scripthash(self):
 
     def is_witness_v0_nested_scripthash(self):
         """Returns true if this is a scriptpubkey for V0 P2WSH embedded in P2SH. """
-        return len(self) == 23 and self[0:2] == b'\xa9\x14' and self[-1] == b'\x87'
+        return len(self) == 23 and self[0:2] == b'\xa9\x14' and self[-1:] == b'\x87'
```
Comment (mcelrath) [inline bitcoin/core/script.py]: These two expressions are identical.  Does the `-1:` change fix some bug?

**Comment by mcelrath** [comment]
Thanks!  I'd be happy to include your patch in my PR.

Did you implement the block parsing functions?  That's the one thing that's missing now before we should get petertodd to merge it.

**Comment by the9ull** [comment]
> Did you implement the block parsing functions? 

This part is missed also from my implementation

**Comment by the9ull** [comment]
The PR was merged but I can't find the commits in `VidaID:segwit`.
What is the problem? Thank you

**Pull request merged**